src/main/java/io/snyk/jenkins/SnykToHTML.java

```diff
@@ -7,64 +7,77 @@
 import io.snyk.jenkins.tools.SnykInstallation;
 import io.snyk.jenkins.transform.ReportConverter;
 import jenkins.model.Jenkins;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.time.Instant;
+import java.io.OutputStream;
+import java.io.PrintStream;
 
-import static io.snyk.jenkins.config.SnykConstants.SNYK_REPORT_HTML;
-import static io.snyk.jenkins.config.SnykConstants.SNYK_TEST_REPORT_JSON;
+import static io.snyk.jenkins.Utils.getURLSafeDateTime;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class SnykToHTML {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnykToHTML.class);
+
   public static FilePath generateReport(
     SnykContext context,
-    SnykInstallation installation
+    SnykInstallation installation,
+    FilePath testJsonPath
   ) {
     try {
       FilePath workspace = context.getWorkspace();
       Launcher launcher = context.getLauncher();
       EnvVars env = context.getEnvVars();
+      PrintStream logger = context.getLogger();
 
-      FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
-      if (!snykReportJson.exists()) {
-        throw new RuntimeException("Snyk Report JSON does not exist.");
+      if (!testJsonPath.exists()) {
+        throw new RuntimeException("Snyk Test JSON does not exist.");
       }
 
-      FilePath reportPath = workspace.child(getURLSafeDateTime() + "_" + SNYK_REPORT_HTML);
-      reportPath.write("", UTF_8.name());
+      FilePath stdoutPath = workspace.child(getURLSafeDateTime() + "_snyk_report.html");
 
-      ArgumentListBuilder args = new ArgumentListBuilder()
+      ArgumentListBuilder command = new ArgumentListBuilder()
         .add(installation.getReportExecutable(launcher))
-        .add("-i", SNYK_TEST_REPORT_JSON);
+        .add("-i", testJsonPath.getRemote());
+
+      int exitCode;
+      try (OutputStream reportWriter = stdoutPath.write()) {
```

Comment:
I forgot to close file handlers here. This `try` will now do that automatically.